waspc/src/Wasp/Lib.hs

```diff
@@ -101,3 +110,17 @@ findMigrationsDir waspDir = do
   let migrationsAbsPath = waspDir SP.</> dbMigrationsDirInWaspProjectDir
   migrationsExists <- doesDirectoryExist $ SP.fromAbsDir migrationsAbsPath
   return $ if migrationsExists then Just migrationsAbsPath else Nothing
+
+loadDockerfileContents :: Path' Abs (Dir WaspProjectDir) -> IO (Maybe Text)
+loadDockerfileContents waspDir = do
+  let dockerfileAbsPath = SP.toFilePath $ waspDir SP.</> [relfile|Dockerfile|]
+  whenMaybeM (doesFileExist dockerfileAbsPath) $ T.IO.readFile dockerfileAbsPath
+
+compileAndRenderDockerfileTemplate :: Path' Abs (Dir WaspProjectDir) -> CompileOptions -> IO (Either [CompileError] Text)
+compileAndRenderDockerfileTemplate waspDir compileOptions = do
```

Comment:
What if there is no dockerfile, what will this function do?

Also, why is it named compileAndRenderDockerfileTemplate, and not just compileAndRenderDockerfile?

Reply:
We should always have a Dockerfile, since we provide it. If there is no user Dockerfile, then we just print ours (with any conditional stuff included too). Or maybe I misunderstand the no Dockerfile scenario?

Yeah, we can nix Template from the name I think. 👍🏻

Reply:
I think I got confused and thought this function works only with the user supplied Dockerfile -> ok, makes sense!

Reply:
Sounds good, no sweat thanks!